driver/conv_driver.hpp

```diff
@@ -439,6 +439,17 @@ int ConvDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
 {
     inflags.Parse(argc, argv);
 
+    // try to set a default layout value for 3d conv if not specified from cmd line
+    int spatial_dim = inflags.GetValueInt("spatial_dim");
+    if(spatial_dim == 3 &&
+       (inflags.GetValueStr("in_layout") == "NCHW" && inflags.GetValueStr("fil_layout") == "NCHW" &&
+        inflags.GetValueStr("out_layout") == "NCHW"))
+    {
```

Comment:
This feels a bit awkward that we use 2d layout try to initialize 3d layout. Since we want either 2d or 3d layout to be initialized with different values, I'd prefer to leave the `in/fil/out_layout` empty when parsing the command line args:

```cpp
bool layout_not_initialized = inflags.GetValueStr("in_layout").empty() &&  inflags.GetValueStr("fil_layout").empty() && inflags.GetValueStr("out_layout").empty() 

if (layout_not_initialized && spatial_dim == 2)
{
    inflags.SetValue("in_layout", "NCHW");
    inflags.SetValue("fil_layout", "NCHW");
    inflags.SetValue("out_layout", "NCHW");
} else if (layout_not_initialized && spatial_dim == 3)
{
    inflags.SetValue("in_layout", "NCDHW");
    inflags.SetValue("fil_layout", "NCDHW");
    inflags.SetValue("out_layout", "NCDHW");
}
```

Reply:
Hi @jerryyin thanks for bring out this comment.
However `inflags.GetValueStr("in_layout").empty()` may not return true since `in/fil/out_layout` will have a default value,  initialized [here]( https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/driver/conv_driver.hpp#L570).
So if we need to set the default value depends on `spatial_dim`, we may need delete the initialization logic in above link. What do you think?

Reply:
@carlushuang Yes that's what I mean. Make all layouts initialized with an empty string, and only set them when we have more information.

Reply:
@jerryyin OK I see. Let me refactor the code based on above discussion.